frame/executive/src/lib.rs

```diff
@@ -564,7 +586,13 @@ where
 
 		// Decode parameters and dispatch
 		let dispatch_info = xt.get_dispatch_info();
-		let r = Applyable::apply::<UnsignedValidator>(xt, &dispatch_info, encoded_len)?;
+		// Check whether we need to error because extrinsics are paused.
+		let r = if ExtrinsicSuspender::is_suspended(dispatch_info.class) {
+			// no refunds
+			Err(DispatchError::Suspended.with_weight(dispatch_info.weight))
```

Comment:
Orthogonal to my other comment but I feel this is wrong: For a non mandatory suspended extrinsic, the extrinsic is not applied but it is noted applied with `note_applied_extrinsic` and the apply extrinsic is a success.

Instead what we want is to not apply suspended extrinsics.
So I think here it should do an early return of `Err(InvalidTransaction::Suspended)` then in client basic-authorship in the proposer we should end the proposing when we find a suspended transaction. (because runtime only accept inherents when it is suspending transactions)
Or we do an early return of `Err(Exhausted)` and the proposer will try `MAX_SKIPPED_TRANSACTIONS=8` other transaction and then end the proposing. ( if I read the code correctly )

(Or we go with making the block full in on_initialize or on_post_inherent)

Reply:
>Instead what we want is to not apply suspended extrinsics.

Yes. This is what i meant above with:  
>Other discussions yielded that we want a runtime API that polls the MBM state and only include non-mandatory transactions into a block when there is no MBM ongoing. But that is more complicated to implement obviously.

This could also be returned as bool from `initialize_block`. So the block builder will then know to not include any extrinsics.  
Do you think that could work? I think it is better than the approach to first include the extrinsic and then reject it from inside the runtime.

Reply:
I changed it to panic now. The block builder should not be able to include non-mandatory transactions while a MBM is ongoing. And a block will not be importable if it does not respect that.